src/module-elasticsuite-core/etc/adminhtml/system.xml

```diff
@@ -180,6 +180,48 @@
+                    <depends>
+                        <field id="extension_enabled">1</field>
+                    </depends>
+                    <comment><![CDATA[Default: No. When set to "Yes", this setting will always proceed to fetch suggestions on the query typed by the user in addition to the extended terms. Eg : When the user is typing "comp", the popular search term "computer" will be suggested, but products or categories matching either "comp" or "computer" will be shown.]]></comment>
+                </field>
+                <field id="extension_limited" translate="label comment" type="select" sortOrder="20" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
+                    <label>Limit the amount of suggested search terms used for autocomplete matching</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <depends>
+                        <field id="extension_enabled">1</field>
+                    </depends>
+                    <comment><![CDATA[Default: No. When set to "No", this setting will use the maximum number of suggested popular search terms configured in the dedicated section.]]></comment>
+                </field>
+                <field id="extension_size" translate="label comment" type="text" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
+                    <label>The number of suggested popular search terms to use for fetching results.</label>
+                    <validate>integer validate-greater-than-zero</validate>
+                    <depends>
+                        <field id="extension_enabled">1</field>
+                        <field id="extension_limited">1</field>
+                    </depends>
+                    <comment><![CDATA[The number of suggested search terms that will be used for fetching results in the autocompletion box.]]></comment>
+                </field>
+                <field id="stop_extension_on_match" translate="label comment" type="select" sortOrder="40" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
+                    <label>Stop the extension mechanism when the current search term of the user is suggested.</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <depends>
+                        <field id="extension_enabled">1</field>
+                    </depends>
+                    <comment><![CDATA[Default: No. When set to "Yes", this setting will discard the extension mechanism as soon as the exact search term entered by the user is part of the suggestions. Eg : When the user has finished to type "computer", if the list of suggested search terms is "computer", "apple computer", "computer for kids", only "computer" will be considered as it's exactly what the user has written.]]></comment>
```

Comment:
I would streamline the phrase like :
When set to "Yes", the extension mechanism will be discarded when search term entered by the user is amongst the popular terms suggestions. Eg : When the user has finished typing "computer", if the list of suggested search terms is ("computer", "apple computer", "computer for kids"), only "computer" will be taken into account for the products and categories autocomplete.